sdk/monitor/monitor-query/review/monitor-query.api.md

```diff
@@ -0,0 +1,313 @@
+## API Report File for "@azure/monitor-query"
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+
+import * as coreHttp from '@azure/core-http';
+import { OperationOptions } from '@azure/core-http';
+import { TokenCredential } from '@azure/core-auth';
+
+// @public
+export interface BatchRequest {
+    requests?: LogQueryRequest[];
+}
+
+// @public
+export interface BatchResponse {
+    error?: BatchResponseError;
+    responses?: LogQueryResponse[];
+}
+
+// @public
+export interface BatchResponseError {
+    code?: string;
+    // (undocumented)
+    innerError?: BatchResponseErrorInnerError;
+    message?: string;
+}
+
+// @public (undocumented)
+export interface BatchResponseErrorInnerError {
+    // (undocumented)
+    code?: string;
+    // (undocumented)
+    details?: ErrorDetails[];
+    // (undocumented)
+    message?: string;
+}
+
+// @public (undocumented)
+export interface ErrorDetails {
+    // (undocumented)
+    code?: string;
+    // (undocumented)
+    message?: string;
+    // (undocumented)
+    target?: string;
+}
+
+// @public
+export const last24Hours = "P1D";
+
+// @public
+export const last2Days = "P2D";
+
+// @public
+export const last30Minutes = "PT30M";
+
+// @public
+export const last3Days = "P3D";
+
+// @public
+export const last48Hours = "P2D";
+
+// @public
+export const last4Hours = "PT4H";
+
+// @public
+export const last5Minutes = "PT5M";
+
+// @public
+export const last7Days = "P7D";
+
+// @public
+export const lastDay = "P1D";
+
+// @public
+export const lastHour = "PT1H";
+
+// @public
+export interface LocalizableString {
+    localizedValue?: string;
+    value: string;
+}
+
+// @public
+export interface LogQueryRequest {
+    body?: QueryBody;
+    headers?: {
+        [propertyName: string]: string;
+    };
+    id?: string;
+    // (undocumented)
+    method?: "POST";
+    // (undocumented)
+    path?: "/query";
+    workspace?: string;
+}
+
+// @public (undocumented)
+export interface LogQueryResponse {
+    body?: QueryResults;
+    // (undocumented)
+    id?: string;
+    // (undocumented)
+    status?: number;
+}
+
+// @public
+export class LogsClient {
+    constructor(tokenCredential: TokenCredential, options?: LogsClientOptions);
+    // (undocumented)
+    queryLogs(workspaceId: string, query: string, options?: QueryLogsOptions): Promise<QueryLogsResult>;
+    // (undocumented)
+    queryLogsBatch(body: BatchRequest, options?: QueryLogsBatchOptions): Promise<QueryBatchResponse>;
```

Comment:
That could be really nice. Java has a nice type and it lets them be a bit more helpful when figuring out which fields make sense to fill out commonly (like workspaceId, rather than workspaces or workspaceIds), and auto-assign an ID, etc.. I'll take a look at data tables.

Reply:
I cleaned this up a bit. I can look at data-tables but the end result the user has to fill out doesn't end up being too bad (we still auto-generated the IDs and take care of ordering, etc..). 

The only "bad" thing is that there are some optional arrays that can be added onto for additional workspace Ids, etc..